### PR TITLE
BFD-1165: upgrade hibernate to 5.6.1

### PR DESCRIPTION
**Pull request by christrotteradhoc**
**JIRA Ticket:**
[BFD-1165](https://jira.cms.gov/browse/BFD-1165)

**User Story or Bug Summary:**

---

### What Does This PR Do?

Upgraded hibernate library as per SNYK recommendation but this resulted in two follow on issues that needed to be resolved:

- The workaround that was in place to override the byte-buddy library version due to a Mockito conflict had to be removed because it was using a scope of test and therefore not getting included in the war file. This appears not to have mattered in the older version of hibernate but the new one is blowing up because of this. The byte-buddy dependency in the new hibernate seems to be compatible with Mockito though -- all tests are passing.
- The new hibernate is not happy about an integer overflow that occurs when paging is not requested due to some code that is adding one to the page size and using that as the max results for a select query. When this code is used with the default page size of Integer.MAX_VALUE, overflow occurs and we end up passing a negative value. The proposed fix is to limit the page size to one less that MAXINT.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/pom.xml [hunk 1]
```diff
@@ -219,15 +219,6 @@
...
 		</dependency>
-		<!-- Hibernate brings in an older version of byte-buddy which seems
-		to conflict with mockito, which wants version 1.11.0.  Bringing it
-		in as a first class dependency to resolve the issue. -->
-		<dependency>
-			<groupId>net.bytebuddy</groupId>
-			<artifactId>byte-buddy</artifactId>
-			<version>1.11.0</version>
-			<scope>test</scope>
-		</dependency>
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/pom.xml:230]: Latest version of hibernate does not have this issue. Leaving this here with the newer version of hibernate leads to other issues so removing this now unnecessary workaround.

**Review by mackec1 — APPROVED** [review]
LGTM, but it feels a little wierd that we don't have properties to define a max page size that we're willing to support.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/PatientLinkBuilder.java [hunk 2]
```diff
@@ -13,6 +13,13 @@
 
 /** A link builder for Patient resources using bene-id cursors */
 public final class PatientLinkBuilder implements LinkBuilder {
+  /**
+   * Maximum page size is one less than the maximum integer value to allow clients to request one
+   * more than the page size as a means to see if an additional page is necessary without having an
+   * integer overflow.
+   */
+  public static final int MAX_PAGE_SIZE = Integer.MAX_VALUE - 1;
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/PatientLinkBuilder.java:21]: 👍

**Review by jzulim — APPROVED** [review]
Nice work!

**Pull request merged**